Update docs.fastly.com links

acme opened the pull request:
Over time the content on https://docs.fastly.com/ has been reorganised and some content has moved to https://developer.fastly.com/. This commit updates the links to point to their new locations.

acme committed:
Update docs.fastly.com links

Over time the content on docs.fastly.com has been reorganised and some
content has moved to developer.fastly.com. This commit updates the
links to point to their new locations.

phamann reviewed (changes requested):
Thank you for this contribution! Very useful. One thing it looks like you've changed the vendored files for `go-fastly` which isn't ideal. Are you able to remove those changes from the PR?

acme committed:
Undo changes to vendored files

The previous commit made changes to vendored files. Changes to those
will be handled separately. This commit undos those changes.

acme commented:
Whoops! I have removed the changes to the vendored files and moved them to https://github.com/fastly/go-fastly/pull/202

phamann reviewed (approved):
:+1: LGTM - Thank you for cleaning this up!
